diff -ur binutils-2.44.debug-dir.orig/gas/read.c binutils-2.44.debug-dir/gas/read.c
--- binutils-2.44.debug-dir.orig/gas/read.c	2025-03-13 03:09:12.160973522 +0000
+++ binutils-2.44.debug-dir/gas/read.c	2025-05-15 19:49:07.416962283 +0000
@@ -43,6 +43,7 @@
 #include "wchar.h"
 #include "filenames.h"
 #include "ginsn.h"
+#include "dwarf2dbg.h"
 
 #include <limits.h>
 
@@ -1546,6 +1547,7 @@
 static void
 s_align (signed int arg, int bytes_p)
 {
+  addressT start_address_for_dwarf2 = frag_now_fix_octets();
   unsigned int align_limit = TC_ALIGN_LIMIT;
   addressT align;
   char *stop = NULL;
@@ -1665,6 +1667,8 @@
 
   if (flag_mri)
     mri_comment_end (stop, stopc);
+
+  dwarf2_emit_insn(frag_now_fix_octets() - start_address_for_dwarf2);
 }
 
 /* Handle the .align pseudo-op on machines where ".align 4" means
@@ -3361,6 +3365,7 @@
 void
 s_space (int mult)
 {
+  addressT start_address_for_dwarf2 = frag_now_fix_octets();
   expressionS exp;
   expressionS val;
   char *p = 0;
@@ -3580,6 +3585,8 @@
 
   if (flag_mri)
     mri_comment_end (stop, stopc);
+
+  dwarf2_emit_insn(frag_now_fix_octets() - start_address_for_dwarf2);
 }
 
 void
@@ -4228,6 +4235,7 @@
 cons_worker (int nbytes,	/* 1=.byte, 2=.word, 4=.long.  */
 	     int rva)
 {
+  addressT start_address_for_dwarf2 = frag_now_fix_octets();
   int c;
   expressionS exp;
   char *stop = NULL;
@@ -4316,6 +4324,8 @@
   if (flag_synth_cfi && frchain_now && frchain_now->frch_ginsn_data
       && nbytes == 1)
     as_bad (_("SCFI: hand-crafting instructions not supported"));
+
+  dwarf2_emit_insn(frag_now_fix_octets() - start_address_for_dwarf2);
 }
 
 void
@@ -5032,6 +5042,7 @@
 float_cons (/* Clobbers input_line-pointer, checks end-of-line.  */
 	    int float_type	/* 'f':.ffloat ... 'F':.float ...  */)
 {
+  addressT start_address_for_dwarf2 = frag_now_fix_octets();
   char *p;
   int length;			/* Number of chars in an object.  */
   char temp[MAXIMUM_NUMBER_OF_CHARS_FOR_FLOAT];
@@ -5106,6 +5117,8 @@
   /* Put terminator back into stream.  */
   --input_line_pointer;
   demand_empty_rest_of_line ();
+
+  dwarf2_emit_insn(frag_now_fix_octets() - start_address_for_dwarf2);
 }
 
 /* LEB128 Encoding.
@@ -5943,6 +5956,7 @@
 void
 stringer (int bits_appendzero)
 {
+  addressT start_address_for_dwarf2 = frag_now_fix_octets ();
   const int bitsize = bits_appendzero & ~7;
   const int append_zero = bits_appendzero & 1;
   unsigned int c;
@@ -6044,6 +6058,8 @@
     }
 
   demand_empty_rest_of_line ();
+
+  dwarf2_emit_insn(frag_now_fix_octets() - start_address_for_dwarf2);
 }
 
 /* FIXME-SOMEDAY: I had trouble here on characters with the
