--- src/vm/errors.nim
+++ src/vm/errors.nim
@@ -140,7 +140,7 @@

     if errMsgParts.len > 1:
         errMsg &= errMsgParts[1..^1].join(fmt("\n{indent}{bold(redColor)}{separator}{resetColor} "))
-    echo fmt("{bold(redColor)}{marker} {header} {separator}{resetColor} {errMsg}")
+    stderr.writeLine fmt("{bold(redColor)}{marker} {header} {separator}{resetColor} {errMsg}")

 #=======================================
 # Methods
