--- /lib/commands.py
+++ /lib/commands.py
@@ -1034,12 +1034,12 @@
 
     delimiter = ''.join(
         random.choice(" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLM"
-                      "NOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~")
+                      "NOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~")
         for _ in range(10))
     while delimiter in a:
         delimiter = ''.join(
             random.choice(" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLM"
-                          "NOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~")
+                          "NOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~")
             for _ in range(10))
 
     a = transliterate(a, b, [delimiter] * len(b))
--- /lib/constants.py
+++ /lib/constants.py
@@ -184,7 +184,7 @@
         arity=0
     ),
     "žQ": MethodAttribute(
-        lambda: " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~",
+        lambda: " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~",
         arity=0
     ),
     "žR": MethodAttribute(
